src/table.js

```diff
@@ -1036,8 +1136,8 @@ Table.prototype.mutate = function(entries, callback) {
   var entryToIndex = new Map(entries.map((entry, index) => [entry, index]));
   var mutationErrorsByEntryIndex = new Map();
 
-  function onBatchResponse(previousNumRequestsMade, err) {
-    if (previousNumRequestsMade === numRequestsMade && err) {
```

Comment:
Was this intentionally  reverted?

Reply:
Yes, although feel free to correct me on this. There was a failing test, and I couldn't understand exactly what was happening (both why we needed the code that I removed & why the test was failing). I attributed it to how we were mocking the endpoints in the test, and now that we switched to GAPIC, the extra code wasn't needed anymore.

Reply:
The issue at hand was that if an error happened without a `request` ever being emitted there would be an infinite (albeit async) infinite loop. This check was a guard against it. The async nature of this method is hiding the fact that it's a glorified `while (true)` loop

Reply:
I see now, thanks. Putting it back.